src/main/java/frc/robot/subsystems/swerve/SwerveDrivetrain.java

```diff
@@ -196,16 +210,25 @@ public void setModuleStates(SwerveModuleState[] states) {
         SmartDashboard.putNumber("Desired angle FL", states[0].angle.getDegrees());
     }
 
+    public void resetGyro(double heading) {
+        m_gyro.setYaw(heading);
+    }
+
     public void resetGyro() {
-        m_gyro.setYaw(0);
+        resetGyro(0);
     }
 
     public Rotation2d getGyroPitch() {
         return Rotation2d.fromDegrees(m_gyro.getPitch());
     }
 
-    public double getGyroRollPitch() {
-        return m_currentRoll - m_previousRoll;
+    public Rotation2d getGyroRoll() {
+        return Rotation2d.fromDegrees(m_gyro.getRoll());
+    }
+
+    public double getGyroPitchRate() {
+        double fpgaElapsedTime = RobotController.getFPGATime() - m_prevTime;
```

Comment:
We need to handle the edge case where m_prevTime == 0. If it is zero, set m_prevTime to the current time.